src/modules/Sidebar/Sidebar.js

```diff
@@ -70,10 +77,40 @@ class Sidebar extends Component {
     this.stopAnimatingTimer = setTimeout(() => this.setState({ animating: false }), duration)
   }
 
+  componentDidMount() {
+    if (this.props.visible && this.props.closable) {
+      document.addEventListener('click', this.handleDocumentClick)
+    }
+  }
+
   componentWillReceiveProps(nextProps) {
     if (nextProps.visible !== this.props.visible) {
       this.startAnimating()
+      if (nextProps.visible && this.props.closable) {
+        document.addEventListener('click', this.handleDocumentClick)
+      } else {
+        document.removeEventListener('click', this.handleDocumentClick)
+      }
+    }
+  }
+
+  handleDocumentClick = (e) => {
+    const sidebarDOMNode = ReactDOM.findDOMNode(this.ref)
+    // close Sidebar if e.target is outside of Sidebar
+    if (!this.checkIfSidebarContainsClickTarget(e.target, sidebarDOMNode)) {
+      this.props.toggleVisibility()
+    }
+  }
+
+  checkIfSidebarContainsClickTarget = (target, node) => {
```

Comment:
I believe method can be replaced with [`node.contains(otherNode)`](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains).  It has been around for a while and has good browser support.